src/test/java/com/example/onlinebookstore/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,67 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.example.onlinebookstore.dto.category.CategoryResponseDto;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.List;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class CategoryControllerTest {
+    protected static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired WebApplicationContext applicationContext
+    ) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @Test
+    @DisplayName("Get all categories")
+    @WithMockUser
+    @Sql(scripts = "classpath:database/"
+            + "categories/add-three-categories.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/"
+            + "categories/delete-all-categories.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void getAll_GivenThreeCategories_ShouldReturnAll() throws Exception {
+        CategoryResponseDto fantasy = new CategoryResponseDto().setId(1L).setName("Fantasy");
+        CategoryResponseDto thriller = new CategoryResponseDto().setId(2L).setName("Thriller");
+        CategoryResponseDto adventure = new CategoryResponseDto().setId(3L).setName("Adventure");
+        List<CategoryResponseDto> expected = List.of(fantasy, thriller, adventure);
+
+        MvcResult result = mockMvc.perform(get("/categories")
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        List<CategoryResponseDto> actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                new TypeReference<List<CategoryResponseDto>>() {});
+        Assertions.assertEquals(expected.size(), actual.size());
+        Assertions.assertEquals(expected, actual);
```

Comment:
what about [assertIterableEquals](https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html#assertIterableEquals-java.lang.Iterable-java.lang.Iterable-)([Iterable](https://docs.oracle.com/javase/8/docs/api/java/lang/Iterable.html?is-external=true)<?> expected, [Iterable](https://docs.oracle.com/javase/8/docs/api/java/lang/Iterable.html?is-external=true)<?> actual)